exp/ipam/api/v1beta1/condition_consts.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+// These reasons are intended to be used with failed Ready conditions on an [IPAddressClaim].
+const (
+	// AllocationFailedReason is the reason used when allocating an IP address for a claim fails. More details should be
```

Comment:
Is there any hint we can give in the comment about user action for this reason, e.g. how permanent / transient this failure might be?